### PR TITLE
Remove all broadcast events

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

Clears the way for the angular upgrade.

medic/cht-core#5624

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/js/directives/actionbar.js [hunk 1]
```diff
@@ -42,6 +67,126 @@ angular.module('inboxDirectives').directive('mmActionbar', function() {
         $state.go('reports.detail', { id: null });
       };
 
+      ctrl.selectAll = () => {
```
Comment (dianabarsan) [inline webapp/src/js/directives/actionbar.js]: Maybe this should be a "reportsAction", along with the other actionbar actions that are connected to reports.

File: webapp/src/js/directives/navigation.js [hunk 2]
```diff
@@ -4,22 +4,41 @@ angular.module('inboxDirectives').directive('mmNavigation', function() {
...
       const unsubscribe = $ngRedux.connect(mapStateToTarget, mapDispatchToTarget)(ctrl);
 
+      /**
+       * Navigate back to the previous view
+       */
+      ctrl.navigateBack = () => {
+        if ($state.current.name === 'contacts.deceased') {
+          $state.go('contacts.detail', { id: $stateParams.id });
+        } else {
+          ctrl.unsetSelected();
```
Comment (dianabarsan) [inline webapp/src/js/directives/navigation.js]: I think removing the additional condition:
```
if ($stateParams.id) {
          $state.go($state.current.name, { id: null });
```
is screwing up tasks.

Comment (garethbowen) [reply webapp/src/js/directives/navigation.js]: I think it was screwed up by a previous commit. Fixed now.

File: webapp/src/js/directives/actionbar.js (hunk 1 above)
Comment (garethbowen) [reply webapp/src/js/directives/actionbar.js]: I've moved this, and everything else in this controller, into the actions. I _think_ this makes sense but it feels strange. Probably just a matter of getting used to it.

**Comment by garethbowen** [comment]
Another review please @dianabarsan , thanks!

**Comment by garethbowen** [comment]
That bug was down to the redux listener firing too often. It's an anti-pattern anyway so I found a way to eliminate it. This code should get cleaner still when we have tasks to disk...

**Comment by dianabarsan** [comment]
I'm seeing this error in the console when navigating away from tasks after previously selecting one:
image

**Comment by garethbowen** [comment]
@dianabarsan Sorry about wasting your time. Please have another look.

**Pull request merged**